modules/subgraphs/veBal-locks-subgraph/veBal-locks-subgraph.service.ts

```diff
@@ -0,0 +1,52 @@
+import { GraphQLClient } from 'graphql-request';
+import { networkContext } from '../../network/network-context.service';
+import _ from 'lodash';
+import { VotingEscrowLock_OrderBy, OrderDirection, getSdk } from './generated/veBal-locks-subgraph-types';
+
+export class VeBalLocksSubgraphService {
+    constructor() {}
+
+    // TODO needs proper paging, currently <3000 locks so it works
```

Comment:
Would not leave this as a todo, just gonna come back and bite you in the ass later, as you'll forget it

Reply:
use `id_gt` as a means to safely paginate